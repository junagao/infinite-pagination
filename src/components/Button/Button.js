import React from 'react'
import { string, bool, func } from 'prop-types'
import ButtonStyled from './Button.styles'

export default function Button({ text, disabled, onClick }) {
  return (
    <ButtonStyled type="button" disabled={disabled} onClick={onClick}>
      {text}
    </ButtonStyled>
  )
}

Button.propTypes = {
  text: string.isRequired,
  disabled: bool.isRequired,
  onClick: func.isRequired,
}
