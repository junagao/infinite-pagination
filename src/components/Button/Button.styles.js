import styled from '@emotion/styled'

const ButtonStyled = styled.button`
  border: ${({ theme }) => theme.borders.none};
  color: ${({ disabled, theme }) =>
    disabled ? theme.colors.disabled : theme.colors.text.body};
  cursor: ${({ disabled }) => (disabled ? 'default' : 'pointer')};
  font-size: ${({ theme }) => theme.fontSizes.xl};
  outline: none;
  padding: ${({ theme }) => `${theme.space[4]} ${theme.space[0]}`};
  pointer-events: ${({ disabled }) => (disabled ? 'none' : 'auto')};
  transition: all 0.2s cubic-bezier(0.645, 0.045, 0.355, 1) 0s;
  width: 100vw;
  &:hover {
    color: ${({ disabled, theme }) =>
      disabled ? theme.colors.disabled : theme.colors.background};
    background-color: ${({ disabled, theme }) =>
      disabled ? theme.colors.background : theme.colors.text.body};
  }
`

export default ButtonStyled
