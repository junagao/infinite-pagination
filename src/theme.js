const theme = {
  colors: {
    background: '#fff',
    primary: '#111',
    text: { body: '#111', hover: '#f0f0f0' },
    disabled: '#ddd',
    error: '#BD271E',
  },
  fontSizes: {
    xl: '4rem',
    lg: '2rem',
    md: '1.5rem',
    sm: '1rem',
    xs: '0.75rem',
  },
  space: [
    0,
    '0.25rem',
    '0.5rem',
    '0.75rem',
    '1rem',
    '1.5rem',
    '2rem',
    '4rem',
    '6rem',
  ],
  borders: {
    none: 'none',
    thin: '1px solid',
  },
  radii: {
    none: 0,
    base: '0.25em',
    round: '99999em',
  },
}

export default theme
