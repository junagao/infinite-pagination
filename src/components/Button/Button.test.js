import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import { ThemeProvider } from 'emotion-theming'
import { matchers } from 'jest-emotion'
import theme from '../../theme'
import Button from './Button'

expect.extend(matchers)

describe('Button', () => {
  test('matches snapshot', () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <Button text="item-6" disabled={false} onClick={jest.fn()} />
      </ThemeProvider>,
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  test('displays the correct item', () => {
    const { getByTestId } = render(
      <ThemeProvider theme={theme}>
        <Button text="item-6" disabled={false} onClick={jest.fn()} />
      </ThemeProvider>,
    )
    expect(getByTestId('button')).toHaveTextContent('item-6')
  })

  test('render disabled button correctly if disabled - set pointer-events to none and color to grey', () => {
    const { getByTestId } = render(
      <ThemeProvider theme={theme}>
        <Button text="item-6" disabled onClick={jest.fn()} />
      </ThemeProvider>,
    )
    expect(getByTestId('button')).toHaveStyleRule('pointer-events', 'none')
    expect(getByTestId('button')).toHaveStyleRule('color', '#ddd')
    expect(getByTestId('button')).toBeDisabled()
  })

  it('fires the onClick method on click', () => {
    const handleClick = jest.fn()
    const { container } = render(
      <ThemeProvider theme={theme}>
        <Button text="item-6" disabled={false} onClick={handleClick} />
      </ThemeProvider>,
    )
    fireEvent.click(container.firstChild)
    expect(handleClick).toHaveBeenCalled()
  })
})
