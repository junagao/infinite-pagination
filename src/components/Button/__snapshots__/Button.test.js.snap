// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches snapshot 1`] = `
.emotion-0 {
  border: none;
  color: #111;
  cursor: pointer;
  font-size: 4rem;
  outline: none;
  padding: 1rem 0;
  pointer-events: auto;
  -webkit-transition: all 0.2s cubic-bezier(0.645,0.045,0.355,1) 0s;
  transition: all 0.2s cubic-bezier(0.645,0.045,0.355,1) 0s;
  width: 100vw;
}

.emotion-0:hover {
  color: #fff;
  background-color: #111;
}

<button
  class="emotion-0 emotion-1"
  data-testid="button"
  type="button"
>
  item-6
</button>
`;
